### PR TITLE
Fix: input alignment & error border for input

### PR DESCRIPTION
**Pull request by parasharrajat**
### Details

Follow up https://github.com/Expensify/App/pull/5742.
1. I have removed the manual translation from the label on ExpensiInput. Previously, It was required to pass custom translateX prop based on the label length. It was hectic and error-prone. 

### Fixed Issues

$ https://github.com/Expensify/App/issues/5462

### Tests | QA Steps
1. Label and value alignment should be correct for input and Select input.
2. When a field has an error it should show a red border. 

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web | Mobile Web | Desktop

Screenshot 2021-10-15 at 02-43-50 New Expensify

#### iOS

Screenshot 2021-10-15 02:44:18

#### Android

Screenshot 2021-10-15 02:43:26

### CODE REVIEW

**Review by aldo-expensify — CHANGES_REQUESTED** [review]
The "focus" border can apply at the same time as the red outline, which looks weird. I think it should be just one border:

Screen Shot 2021-10-13 at 9 37 43 AM

**Comment by shawnborton** [comment]
I think we want to keep the :focus styles for accessibility purposes right?

**Comment by shawnborton** [comment]
But that being said, maybe we want the :focus style on an input with an error border to look different? It seems like a pretty standard :focus outline is to take the :focus color, reduce the opacity, and apply it as a thicker drop shadow (with no blur) around the input.. GH for example: 
image

**Comment by shawnborton** [comment]
@parasharrajat in this screenshot here:
image


It feels like the "He/him" is over to the right by 1px too far. Can you inspect that one for me and show me the bounding boxes?

File: src/styles/styles.js
Comment (parasharrajat) [reply src/styles/styles.js:542]: Oh, I forget that point while figuring out other issues. I will update it.

**Comment by parasharrajat** [comment]
Sorry for waiting. I will fix this up in 2 hours.

**Comment by shawnborton** [comment]
No worries, just let me know when it's ready for another review.

**Comment by aldo-expensify** [comment]
> Changes pulled I don't think anything is broken on AdressSearch. Could you please verify that?

It looks fine:

Screen Shot 2021-10-14 at 3 32 09 PM

One thing that looks a bit off is the color for picker inputs:
Screen Shot 2021-10-14 at 3 33 11 PM

There is like a bit of shadow I think

File: src/styles/styles.js
Comment (shawnborton) [inline src/styles/styles.js:535]: This should just be 11 and not 11.5

Comment (shawnborton) [reply src/styles/styles.js:535]: Ah sorry for the confusion. We want 11 on each side and not 11.5. Thanks for fixing!

**Comment by parasharrajat** [comment]
Hmm. I will check the border thing. I have to look back at the old PR to know why I did this and a couple of tests to make sure that it works without issue.

I can open a different PR for that if you think this PR is ready.

File: src/styles/styles.js
Comment (shawnborton) [reply src/styles/styles.js:535]: I think you forgot about this one @parasharrajat

**Comment by parasharrajat** [comment]
Fixed @shawnborton . Thanks for reminding me.

**Review by aldo-expensify — CHANGES_REQUESTED** [review]
Tested on web and it was looking fine.

There is a problem in Android, inputs are loosing the right border:

Screenshot_20211020-193610

**Comment by parasharrajat** [comment]
That's unfortunate. Checking now.

**Comment by parasharrajat** [comment]
>  Why aren't we just using a hard-coded font size in both places? I don't think we have to change it as the font size appears to be the same, but was just curious.

I tell you that scaling is making making everything complex here but it is needed. On TextInput when there is no text, the label will be placed in the center, and the size of it would be bigger than the active label. so to animate this behavior we scale it down.

**Comment by parasharrajat** [comment]
Hey, @shawnborton Checking this now.
I see no difference, could you please check again, I can access the VBA page?
Screenshot 2021-11-04 22:09:17

**Comment by parasharrajat** [comment]
Any thoughts?

**Comment by shawnborton** [comment]
I think this looks better! The best way to test though is of course to use the same label + value for each, and then I can overlay one on top of the other to see if they are off at all. But I think we're getting close!

**Comment by parasharrajat** [comment]
I don't know why I am holding this for no reason. I will share the screenshots soon and recheck if something changes after merging the main.

**Comment by parasharrajat** [comment]
Experiment with the new Input design and label placement so that there is no need to manually adjust the label for Android on Each page. 
I come to know that the current design of Input needs to define the label Placement Explicitly for each page. it depends on the length of the text on the label. That's why we keep getting the issue back on different pages.

**Comment by parasharrajat** [comment]
Ok. This is ready now. cc: @shawnborton @aldo-expensify. 
There has been a major rewrite of the BaseExpensiTextInput to remove the need of manually placing the label position for each use of it.
It makes it more robust and error-free.
I cleaned up a lot of code and Please let me know if more code can be removed.

**Comment by aldo-expensify** [comment]
Thanks  @parasharrajat , I'll review this as soon as I finish with something I'm working on!

**Comment by parasharrajat** [comment]
Thanks, @aldo-expensify, please take your time. FYI, I am expecting regression testing from your review. I can't access the VBA flow. So if you could test that, it would help me fix the input once for all.

**Review by aldo-expensify — CHANGES_REQUESTED** [review]
The code `const pickerProps = _.omit(this.props, _.keys(propTypes));` in `ExpensiPicker` is omitting the key `hasError`, then `Picker` never gets it and there is no red outline.

**Comment by parasharrajat** [comment]
Fixing it.

**Comment by parasharrajat** [comment]
@aldo-expensify Changes done.

**Review by aldo-expensify — DISMISSED** [review]
Thanks for fixing that @parasharrajat 

Tested on Android and Web and seems to be working fine.

File: src/components/ExpensiTextInput/index.native.js
Comment (aldo-expensify) [inline src/components/ExpensiTextInput/index.native.js:11]: NAB: I see that this file changed from index.ios.js to index.native.js, so it is now used for IOS and Android, but the comment is only about Android throwing an error now. iOS doesn't throw this error?

If both throw this error, we should update the comment to say something like iOS/Android

Comment (parasharrajat) [reply src/components/ExpensiTextInput/index.native.js:11]: YUP 😁. I will update.

**Review by aldo-expensify — DISMISSED** [review]
Looks good to me!

**Comment by parasharrajat** [comment]
@shawnborton @MariaHCD  Awaiting your review. I am not sure who will merge it though.

**Comment by shawnborton** [comment]
I'll give this another review today.

**Comment by shawnborton** [comment]
Alignment-wise, this looks pretty good to me! Nice job, Rajat. 

Otherwise, curious where we landed with the :outline style conversation?

**Comment by parasharrajat** [comment]
So there won't be any alignment issues in the future. All inputs are now calculating their own label placement position so I am sure it would be 99.999% perfect. 
For outline issues that Aldo mentioned
* https://github.com/Expensify/App/pull/5805#pullrequestreview-778836004 Fixed! only red is shown now if the input has an error.
* https://github.com/Expensify/App/pull/5805#issuecomment-943825866 Fixed! Previously both gray and red were shown together now only one is shown (blue, red, or gray) based on the input state.

Please let me know if there is something remaining. This PR thread has so many comments I could have missed some points. But I think fixed all along the way.

I didn't expect that this issue would take this much time as I didn't think the original input implementation has issues and used the wrong props (use of translate props as it is). Now I decided to refactor it.

**Comment by shawnborton** [comment]
Sounds good to me!

**Review by shawnborton — DISMISSED** [review]

**Comment by parasharrajat** [comment]
Maria is in a different time zone. Please give it :green_circle: 🚦 for merge.

**Comment by shawnborton** [comment]
I think it's okay if Maria merges this tomorrow once she sees it and gets a chance to review. Thanks for your patience!

**Comment by parasharrajat** [comment]
@MariaHCD Please review it.

**Comment by parasharrajat** [comment]
I don't think Maria is available. I see no other reason for holding this. Just not looking for more merge conflicts. cc: @aldo-expensify

**Comment by shawnborton** [comment]
Yup - looks like Maria is OOO this week, so I think we can go ahead and merge.

**Comment by parasharrajat** [comment]
Well...well...well. Look what we got here....conflicts 🤦‍♂️

**Comment by aldo-expensify** [comment]
> Well...well...well. Look what we got here....conflicts 🤦‍♂️

:( :( :( :( thanks @parasharrajat , had a laugh with your gif, it was perfect 😆

**Review by shawnborton — APPROVED** [review]

**Review by aldo-expensify — APPROVED** [review]

**Comment by aldo-expensify** [comment]
I guess we wait for the E2E tests and then merge?

**Comment by aldo-expensify** [comment]
Merging, @MariaHCD is OOO this week

File: src/components/ExpensiTextInput/ExpensiTextInputLabel/index.native.js
Comment (eVoloshchak) [inline src/components/ExpensiTextInput/ExpensiTextInputLabel/index.native.js:29]: Dropping a friendly note that this has caused a regression in https://github.com/Expensify/App/issues/17715. 
There is a detailed explanation [here](https://github.com/Expensify/App/issues/17715#issuecomment-1519917598), but in short, on Android, `onLayout` was firing after the screen animation, setting `this.state.width` and moving thelabel. This resulted in labels jumping horizontally right after the screen animation on Android

**Pull request merged**
